### PR TITLE
A workaround for #1377

### LINKED ISSUE #1377: Main custom episode naming format (CENF) is affected by sub-option for unrelated CENF
I came across this issue after migrating from SickBeard. I copied the pattern verbatim from an old SickBeard installation.

The following pattern:

```
Season %S/%SN - Season %S, Episode - %EN/%RN
```

renders correctly as:

```
Season 2/Show Name - Season 2, Episode - Ep Name/Show.Name.S02E03.HDTV.XviD-RLSGROUP.ext
```

But, by adding in one parameter,:

```
Season %S/%SN - Season %S, Episode %E - %EN/%RN
```

the season/episode format is completely lost from the folder name, apparently being interpreted together with the new parameter as an anime (or DVD?)-ordering value. This is rendered as:

```
Season 2/Show Name - 003 - Ep Name/Show.Name.S02E03.HDTV.XviD-RLSGROUP.ext
```

Am I doing something wrong?

### LINKED ISSUE #2545: Config - Post-processing - Naming rule
SickRage Info:
Branch: master
Commit: 8bf53e8d03dc2b5bb946123dbaef2afb8667730c
Version: v2016.10.20-1
Database Version: 44.0
  Version de Python: 2.7.9 (default, Sep 17 2016, 20:26:04) [GCC 4.9.2]
  Version SSL: OpenSSL 1.0.1t 3 May 2016
  OS: Linux-4.4.13-v7+-armv7l-with-debian-8.0
#   Local: fr_FR.UTF-8

Hello,

I use a custom naming rule for Series episode.
if I want to use a custom naming rule for anime, this impact custom naming rulefor Series.

We could not differentiate between the two?

Example:

for serie : custom naming rule : Season %0S/%S_N_-_S%0SE%0E_-_%E_N_-_[%Q_N]
For anime I would  : %S_N_-_EP%AB_-_%E_N_-_[%Q_N]

### PR DESCRIPTION
**Pull request by sharkykh**
Fixes #1377

When trying to use an episode naming pattern in the following scenario:

- **Pattern:** `Season %S/%SN - Season %S, Episode %E - %EN/%RN` (_notice there are two directories_)
- **Custom Anime Naming:** Selected (_otherwise known as NAMING_CUSTOM_ANIME=1_)
- **Anime Absolute Number:** [Add/Only](https://cloud.githubusercontent.com/assets/10238474/23103252/e149c0f6-f6c0-11e6-8d7b-1a01c4051dfe.png) (_otherwise known as anime_type=1 or anime_type=2_)

You would get
`Season 2/Show Name - 003 - Ep Name/Show.Name.S02E03.HDTV.XviD-RLSGROUP.ext`
**for a normal (non-anime) episode**,
instead of
`Season 2/Show Name - S02E03 - Ep Name/Show.Name.S02E03.HDTV.XviD-RLSGROUP.ext`

This is a 'hacky' fix, a workaround for this problem.
Any suggestions for a better one are welcome.

EDIT:
Looks like it will also fix #2545.

### CODE REVIEW

**Comment by SickRage2** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked

- [x] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

File: sickbeard/tv.py [hunk 1]
```diff
@@ -2391,7 +2391,9 @@ def formatted_dir(self, pattern=None, multi=None):
         if len(name_groups) == 1:
             return ''
         else:
-            return self._format_pattern(os.sep.join(name_groups[:-1]), multi)
```
Comment (h3llrais3r) [inline sickbeard/tv.py:2394]: I think this line should not be changed.
Instead, the anime_type should also be added here:
https://github.com/sharkykh/SickRage/blob/develop/sickbeard/naming.py#L208

Can you try again with these changes?
By doing it this way, we only pass the anime_type to the formatted_dir function and it seems to give the correct results.

@miigotu what do you think?

**Comment by sharkykh** [comment]
Let me see if I understand you correctly... https://github.com/sharkykh/SickRage/commit/68903347b3414999ee23c74adc874e603c1733d5 (this is based off a clean develop again)

Result:
Worked for testing at the Episode Naming config,
but failed in Preview Rename of a show.

**Comment by h3llrais3r** [comment]
You're right. I'm checking the preview rename now...

**Comment by h3llrais3r** [comment]
Replace https://github.com/SickRage/SickRage/pull/3178/files#diff-28cfafa47039ed0a3b3f6be8687d0811R2368 by the line below and it should be fine now.
``
result = ek(os.path.join, self.formatted_dir(anime_type=anime_type), result)
``

**Review by h3llrais3r — CHANGES_REQUESTED** [review]
> Let me see if I understand you correctly... sharkykh@6890334 (this is based off a clean develop again)

Replace https://github.com/sharkykh/SickRage/blob/develop/sickbeard/tv.py#L2368 by the line below and it should be fine now.
``
result = ek(os.path.join, self.formatted_dir(anime_type=anime_type), result)
``

File: sickbeard/tv.py (hunk 1 above)
Comment (sharkykh) [reply sickbeard/tv.py:2394]: Just checked, it works perfectly.
Updated the PR.
Thanks @h3llrais3r :)

**Review by h3llrais3r — APPROVED** [review]

**Comment by h3llrais3r** [comment]
@miigotu Can you do a final review?

File: sickbeard/tv.py (hunk 1 above)
Comment (h3llrais3r) [reply sickbeard/tv.py:2394]: Thanks for the PR 😄

**Comment by miigotu** [comment]
Fine with me =P
Approved

[Approved with PullApprove](https://pullapprove.com/SickRage/SickRage/pull-request/3178/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=SickRage/SickRage)

**Pull request merged**